### PR TITLE
Add case re-assign test scenarios

### PR DESCRIPTION
**Pull request by nasirkhanbeis**
Add case re-assign test scenarios
## Description
Added test scenarios around case re-assign within same team and someone else out side team

## Checklist:

- [ ] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/spec/features/assign_investigation_spec.rb [hunk 1]
```diff
@@ -21,4 +24,35 @@
...
+  end
+
+  scenario "assign case to someone else in another team" do
+    visit new_investigation_assign_path(investigation)
+    choose("Someone else")
+    select another_active_user_another_team.name, from: "investigation_select_someone_else"
+    click_button "Continue"
+    fill_in "investigation_assignee_rationale", with: "Test assign"
+    click_button "Confirm change"
+    expect(page).to have_css(".govuk-summary-list__value", text: another_active_user_another_team.name.to_s)
```
Comment (slorek) [inline psd-web/spec/features/assign_investigation_spec.rb]: Try to assert that this corresponds to the assignee, for instance in other tests I used the `have_sibling` assertion, e.g.     `expect(page.find("th", text: "Title")).to have_sibling("td", text: document.title)`.

Comment (nasirkhanbeis) [reply psd-web/spec/features/assign_investigation_spec.rb]: Sure @slorek will use for table elements, for above page I don't see it in table structure

Comment (frankieroberto) [reply psd-web/spec/features/assign_investigation_spec.rb]: @nasirkhanbeis @slorek I created a custom matcher in the cosmetics project for asserting key-value pairs in "Check your answers" summaries: https://github.com/UKGovernmentBEIS/beis-opss/blob/master/cosmetics-web/spec/support/matchers/capybara_matchers.rb#L11 – it currently expects table markup but can (and should) be updated to expect `<dl>` markup as we convert the templates.

File: psd-web/spec/features/assign_investigation_spec.rb [hunk 2]
```diff
@@ -5,12 +5,15 @@
...
+  let!(:another_active_user) { create(:user, :activated, name: "other user same team", organisation: user.organisation, teams: [team]) }
   let!(:another_inactive_user) { create(:user, :inactive, organisation: user.organisation, teams: [team]) }
-  let!(:another_active_user_another_team) { create(:user, :activated, organisation: user.organisation, teams: [create(:team)]) }
+  let!(:another_active_user_another_team) { create(:user, :activated, name: "another user in another team", organisation: user.organisation, teams: [create(:team)]) }
   let!(:another_inactive_user_another_team) { create(:user, :inactive, organisation: user.organisation, teams: [create(:team)]) }
 
-  before { sign_in(as_user: user) }
+  before {
+    sign_in(as_user: user)
+  }
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/assign_investigation_spec.rb]: bit nit picky but one line when using `{}` or use the `do/end` for multi lines

File: psd-web/spec/features/assign_investigation_spec.rb (hunk 1 above)
Comment (StupidCodeFactory) [reply psd-web/spec/features/assign_investigation_spec.rb]: I did the same

File: psd-web/spec/features/assign_investigation_spec.rb (hunk 2 above)
Comment (nasirkhanbeis) [reply psd-web/spec/features/assign_investigation_spec.rb]: fixed

File: psd-web/spec/features/assign_investigation_spec.rb (hunk 1 above)
Comment (nasirkhanbeis) [reply psd-web/spec/features/assign_investigation_spec.rb]: I think miss understood earlier, fixed now! @slorek

**Pull request merged**